pkg/client/client_test.go

```diff
@@ -15,3 +21,136 @@ func TestGetSchemeHostKey(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, expected, actual)
 }
+
+func TestRetryPolicy(t *testing.T) {
+	bgCtx := context.Background()
+	chCtx := context.WithValue(bgCtx, config.ConsistentHashingStrategyKey, true)
+	errContext, cancel := context.WithCancel(bgCtx)
+	cancel()
+
+	urlError := &url.Error{Err: fmt.Errorf("stopped after 15 redirects"), URL: "http://example.com"}
+
+	tc := []struct {
+		name           string
+		ctx            context.Context
+		resp           *http.Response
+		err            error
+		expectedResult bool
+		expectedError  error
+	}{
+		{
+			name:           "context error",
+			ctx:            errContext,
+			resp:           &http.Response{},
+			err:            context.Canceled,
+			expectedResult: false,
+			expectedError:  context.Canceled,
+		},
+		{
+			name:           "net.OpErr: dial",
+			ctx:            chCtx,
+			resp:           &http.Response{},
+			err:            &net.OpError{Op: "dial"},
+			expectedResult: false,
+			expectedError:  client.ErrStrategyFallback,
+		},
+		{
+			name:           "net.OpErr: read",
+			ctx:            chCtx,
+			resp:           &http.Response{},
+			err:            &net.OpError{Op: "read"},
+			expectedResult: false,
+			expectedError:  client.ErrStrategyFallback,
+		},
+		{
+			name:           "net.DNSErr: Timeout",
+			ctx:            chCtx,
+			resp:           &http.Response{},
+			err:            &net.DNSError{IsTimeout: true},
+			expectedResult: false,
+			expectedError:  client.ErrStrategyFallback,
+		},
+		{
+			name:           "net.DNSErr: IsTemporary",
+			ctx:            chCtx,
+			resp:           &http.Response{},
+			err:            &net.DNSError{IsTemporary: true},
+			expectedResult: false,
+			expectedError:  client.ErrStrategyFallback,
+		},
+		{
+			name:           "net.ErrClosed",
+			ctx:            chCtx,
+			resp:           &http.Response{},
+			err:            net.ErrClosed,
+			expectedResult: false,
+			expectedError:  client.ErrStrategyFallback,
+		},
+		{
+			name:           "Unrecoverable error",
+			ctx:            chCtx,
+			resp:           &http.Response{},
+			err:            urlError,
+			expectedResult: false,
+		},
+		{
+			name:           "Status Bad Gateway",
+			ctx:            chCtx,
+			resp:           &http.Response{StatusCode: http.StatusBadGateway},
+			expectedResult: false,
+			expectedError:  client.ErrStrategyFallback,
+		},
+		{
+			name:           "Status OK",
+			ctx:            chCtx,
+			resp:           &http.Response{StatusCode: http.StatusOK},
+			err:            urlError,
+			expectedResult: false,
+		},
+		{
+			name:           "Status Service Unavailable",
+			ctx:            chCtx,
+			resp:           &http.Response{StatusCode: http.StatusBadGateway},
+			expectedResult: false,
+			expectedError:  client.ErrStrategyFallback,
+		},
+		{
+			name:           "Recoverable Error",
+			ctx:            chCtx,
+			resp:           &http.Response{StatusCode: http.StatusTooManyRequests},
```

Comment:
```suggestion
			resp:           &http.Response{StatusCode: http.StatusOK},
```
otherwise this is confusing with the next case